### PR TITLE
Angular: Add built-in Ivy support instead of relying on addon

### PR DESCRIPTION
**Pull request by kroeder**
Issue:

## What I did

This PR integrates storybook-angular-addon-ivy into
the mono-repo. This allows us to remove the ivy addon
package.

## How to test

tba

### CODE REVIEW

File: app/angular/src/server/framework-preset-angular-ivy.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export const webpack = async (webpackConfig: Configuration, options: Options) => {
+  const angularOptions = await options.presets.apply(
+    'angularOptions',
+    {} as {
+      enableIvy?: boolean;
+    },
+    options
+  );
+
+  if (!angularOptions.enableIvy) {
```
Comment (kroeder) [inline app/angular/src/server/framework-preset-angular-ivy.ts:32]: Should actually default to "true" if user-land has not passed in `enableIvy`

**Review by shilman — CHANGES_REQUESTED** [review]
LGTM except for:
- [x] Fixing the default value handling
- [ ] documentation

File: app/angular/src/server/framework-preset-angular-ivy.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Run ngcc for converting modules to ivy format before starting storybook
+ * This step is needed in order to support Ivy in storybook
+ *
+ * Information about Ivy can be found here https://angular.io/guide/ivy
+ */
+export const runNgcc = () => {
+  ngccProcess({
+    // should be async: true but does not work due to
```
Comment (ThibaudAV) [inline app/angular/src/server/framework-preset-angular-ivy.ts:14]: Shouldn't we change this or should we wait ?

Comment (kroeder) [reply app/angular/src/server/framework-preset-angular-ivy.ts:14]: It does work without being async. The downside is if someone starts the user app + storybook at the same time and both start `ngcc` then storybook will most likely crash 

So it's an edge case but it also does not work yet to set it to async

File: app/angular/src/server/framework-preset-angular-ivy.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const angularOptions = await options.presets.apply(
+    'angularOptions',
+    {} as {
+      enableIvy?: boolean;
+    },
+    options
+  );
+
+  // Default to true, if undefined
+  if (angularOptions.enableIvy === false) {
```
Comment (kroeder) [inline app/angular/src/server/framework-preset-angular-ivy.ts:33]: @shilman changing it to `=== false` should do the trick I guess?

Only noop if explicitly `enableIvy` is set to false in user-land

Comment (shilman) [reply app/angular/src/server/framework-preset-angular-ivy.ts:33]: LGTM!

**Comment by kroeder** [comment]
@shilman I added docs and successfully tested it locally with `npm link`

**Comment by kroeder** [comment]
@shilman any idea why the e2e-tests-core step fails?

**Comment by shilman** [comment]
@kroeder  This is falling on next also. @gaetanmaisse is on top of it

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**